newIDE/app/src/BehaviorsEditor/Editors/BehaviorPropertiesEditor.js

```diff
@@ -7,30 +7,78 @@ import propertiesMapToSchema from '../../PropertiesEditor/PropertiesMapToSchema'
+import Text from '../../UI/Text';
+import { type Schema } from '../../PropertiesEditor';
 
 type Props = BehaviorEditorProps;
 
 export default class BehaviorPropertiesEditor extends React.Component<Props> {
+  _getPropertyGroupNames = (): Array<string> => {
+    const { behavior, behaviorContent } = this.props;
+    const properties = behavior.getProperties(behaviorContent.getContent());
+
+    const groupNames = new Set<string>();
+    const propertyNames = properties.keys();
+    for (let i = 0; i < propertyNames.size(); i++) {
+      const name = propertyNames.at(i);
+      const property = properties.get(name);
+      const group = property.getGroup() || '';
+      groupNames.add(group);
+    }
+    return [...groupNames].sort((a, b) => a.localeCompare(b));
+  };
+
   render() {
     const { behavior, behaviorContent, object } = this.props;
     const properties = behavior.getProperties(behaviorContent.getContent());
 
-    const propertiesSchema = propertiesMapToSchema(
-      properties,
-      behaviorContent => behavior.getProperties(behaviorContent.getContent()),
-      (behaviorContent, name, value) => {
-        behavior.updateProperty(behaviorContent.getContent(), name, value);
-      },
-      object
-    );
+    const groupNames = this._getPropertyGroupNames();
+
+    let schemaIsEmpty = true;
+    const propertiesSchemas: Schema[] = groupNames.map(groupName => {
+      const propertiesSchema = propertiesMapToSchema(
+        properties,
+        behaviorContent => behavior.getProperties(behaviorContent.getContent()),
+        (behaviorContent, name, value) => {
+          behavior.updateProperty(behaviorContent.getContent(), name, value);
+        },
+        object,
+        groupName
+      );
+      schemaIsEmpty = schemaIsEmpty && !propertiesSchema;
+      return propertiesSchema;
+    });
 
     return (
       <Column expand>
-        {propertiesSchema.length ? (
-          <PropertiesEditor
-            schema={propertiesSchema}
-            instances={[behaviorContent]}
-          />
+        {!schemaIsEmpty ? (
+          <ColumnStackLayout key={behaviorContent.getName()} noMargin>
+            {propertiesSchemas.map((propertiesSchema, index) => {
+              if (!propertiesSchema) {
+                return null;
+              }
+              if (!groupNames[index]) {
+                return (
+                  <PropertiesEditor
+                    key={groupNames[index]}
+                    schema={propertiesSchema}
+                    instances={[behaviorContent]}
+                  />
+                );
+              }
+              return [
+                <Text key={groupNames[index] + 'title'} size="title">
+                  {groupNames[index]}
+                </Text>,
+                <PropertiesEditor
+                  key={groupNames[index]}
+                  schema={propertiesSchema}
+                  instances={[behaviorContent]}
+                />,
```

Comment:
Do you think we could put this logic of the title rendering inside the schema and PropertiesEditor?

Probably when converting the properties to a Schema, we can enhance the existing type of field containing other fields:

![image](https://user-images.githubusercontent.com/1280130/146637793-e08b8da2-7030-4e2a-b2d7-38670ba0f55b.png)

and render the title here:
![image](https://user-images.githubusercontent.com/1280130/146637832-6ea69527-37e5-45a7-a3de-39f72331eb80.png)


This needs some logic when converting the properties to, well, group them inside the returned Schema :) The advantage is that this is not specific to behaviors and could be used for objects properties, instance properties and other places where properties are shown (the resource editor)